### PR TITLE
Refactor all operations to use app_id when present.  Ultimately to ac…

### PR DESCRIPTION
**Pull request by jaleksynas**
…count for ITC users that do not have admin security rights.

Requires spaceship PR 265 (https://github.com/fastlane/spaceship/pull/265)
#85

### CODE REVIEW

File: lib/pilot/tester_manager.rb [hunk 1]
```diff
@@ -6,33 +6,26 @@ class TesterManager < Manager
...
-            app = Spaceship::Application.find(app_filter)
-            raise "Couldn't find app with '#{app_filter}'" unless app
-            tester.add_to_app!(app.apple_id)
-            Helper.log.info "Successfully added tester to app #{app_filter}".green
-          rescue => ex
-            Helper.log.error "Could not add #{tester.email} to app: #{ex}".red
-            raise ex
-          end
-        end
+      tester = Spaceship::Tunes::Tester::Internal.find_by_app(app_id, config[:email])
```
Comment (lacostej) [inline lib/pilot/tester_manager.rb]: These two lines could be extracted into one method. They are reused in a few places.

File: lib/pilot/tester_manager.rb [hunk 2]
```diff
@@ -6,33 +6,26 @@ class TesterManager < Manager
...
+                                                            app_id: app_id)
+        Helper.log.info "Successfully invited tester: #{tester.email}".green
+      end
+
+      begin
+        tester.add_to_app!(app_id) if app_id
+        Helper.log.info "Successfully added tester to app #{app_id}".green
       rescue => ex
-        Helper.log.error "Could not create tester #{config[:email]}".red
+        Helper.log.error "Could not add #{tester.email} to app: #{ex}".red
```
Comment (lacostej) [inline lib/pilot/tester_manager.rb]: Any reason the overall `begin/rescue` disappeared ?

File: lib/pilot/tester_manager.rb [hunk 3]
```diff
@@ -139,5 +132,14 @@ def describe_tester(tester)
         rows: rows
       )
     end
+
+    def find_app_id
```
Comment (lacostej) [inline lib/pilot/tester_manager.rb]: maybe document a bit. Returns `nil` if no apple_id or app_identifier specified.

**Comment by mpirri** [comment]
@jaleksynas same deal as with https://github.com/fastlane/spaceship/pull/265, how much of this do you want to tackle before we do the mono-repo changeover?

Incidentally, one of the reasons we are looking forward to the mono-repo changeover is that paired PRs like this one and https://github.com/fastlane/spaceship/pull/265 will only need to be one PR!

**Comment by jaleksynas** [comment]
Refactored accordingly.  There is still some attention due to the various app_id lookup methods in `manager.rb`.  I will take that on after mono-repo migration.

**Comment by mpirri** [comment]
@jaleksynas We are getting ready for the mono repo change on our end -- anything else that needs to your PR on `spaceship` to be able to merge this before that changeover happens?  Thanks!

**Comment by jaleksynas** [comment]
If they everyone is happy with the changes both PRs seem good to me.

**Comment by mpirri** [comment]
@jaleksynas I am so sorry to do this to you, but we didn't quite get through all of our validation on your PRs before our mono-repo transition deadline this morning.  As such, do you mind combining these two PRs into a single PR and resubmitting tomorrow on the new repository?  I expect this process should be pretty straightforward, but if you need help with it, we're happy to lend a hand - just let us know.  Thanks again for your hard work, and I apologize for this inconvenience!

In the meantime, I'm going to close this PR and also https://github.com/fastlane/spaceship/pull/265

**Pull request closed without merging**